src/Eloquent/Traits/HasAdjacencyList.php

```diff
@@ -147,6 +147,7 @@ public function ancestors()
      */
     public function ancestorsAndSelf()
     {
+        // @phpstan-ignore return.type
```

Comment:
This can be removed if you change return type to:
```diff
-     * @return \Staudenmeir\LaravelAdjacencyList\Eloquent\Relations\Ancestors<self>
+     * @return \Staudenmeir\LaravelAdjacencyList\Eloquent\Relations\Ancestors<static>
```